Quotient/events/roommessageevent.h

```diff
@@ -46,24 +46,45 @@ class QUOTIENT_API RoomMessageEvent : public RoomEvent {
     MsgType msgtype() const;
     QString rawMsgtype() const;
     QString plainBody() const;
-    const EventContent::TypedBase* content() const { return _content.get(); }
+
+    //! \brief The EventContent for this event.
+    //!
+    //! \warning The result must be checked for nullptr as an event with just a plainBody
+    //!          will not have a content object.
+    //!
+    //! \return an EventContent object if the event has content, nullptr otherwise.
+    std::unique_ptr<EventContent::TypedBase> content() const;
+
     void editContent(auto visitor)
     {
-        visitor(*_content);
-        editJson()[ContentKey] = assembleContentJson(plainBody(), rawMsgtype(), _content.get(), _relatesTo);
+        visitor(*content());
+        editJson()[ContentKey] = assembleContentJson(plainBody(), rawMsgtype(), content().get(), relatesTo());
```

Comment:
That won't work: the value returned by `content()` is a temporary and gets deleted after the `visitor()` invocation. What you want is
```suggestion
        auto c = content();
        visitor(*c);
        editJson()[ContentKey] = assembleContentJson(plainBody(), rawMsgtype(), c.get(), relatesTo());
```
And to be honest, if we don't store the content object then I would replace `editContent()` with simple `setContent()` that accepts `std::unique_ptr<EventContent::TypedBase>` and only consists of the `editJson()...` line, and push the content "visit" out of the class entirely. E.g., the code that uses `editContent()` in `eventitem.cpp` would then look like:
```cplusplus
        auto fc = rme->fileContent();
        fc->source = uploadedFileData;
        rme->setContent(std::move(fc));
```